com.unity.shadergraph/Editor/Drawing/Controllers/BlackboardController.cs

```diff
@@ -5,14 +5,53 @@
 using UnityEngine.UIElements;
 using System;
 using UnityEditor.Graphing;
-using UnityEditor.ShaderGraph.Drawing.Views;
 using UnityEditor.ShaderGraph.Internal;
-using UnityEngine.Assertions;
 using GraphDataStore = UnityEditor.ShaderGraph.DataStore<UnityEditor.ShaderGraph.GraphData>;
 using BlackboardItem = UnityEditor.ShaderGraph.Internal.ShaderInput;
 
 namespace UnityEditor.ShaderGraph.Drawing
 {
+    struct BlackboardShaderInputOrder
+    {
+        public bool isKeyword;
+        public KeywordType keywordType;
+        public ShaderKeyword builtInKeyword;
+        public string deprecatedPropertyName;
+        public int version;
+    }
+    class BlackboardShaderInputFactory
+    {
+        static public ShaderInput GetShaderInput(BlackboardShaderInputOrder order)
+        {
+            ShaderInput output;
+            if (order.isKeyword)
```

Comment:
Does this need handling for dropdowns?
else if (order.isDropdown)...

Reply:
If I drag properties from one subgraph to another, the properties will often not get copied. I'm not sure if this is new but it looks like it only works if you release the mouse button over the top of one of the existing properties?

Reply:
Duplicating properties does not appear to work any more. On the blackboard, right-click a properties, Duplicate -> nothing appears to be duplicated.

Reply:
Yup this is missing handling for the dropdowns, missed it during merging in from master cause it "just worked"

Will also look at why they dont get added to selected categories 👍

Reply:
Landon caught the breakage of dragging properties/categories between graphs, I didn't think to handle that, will address it in the next iteration

The two categories reversing bug is new, good catch, will look into that

Reply:
These bugs are handled now, and have taken into account dropdowns in this segment of the code